### PR TITLE
Feature/update jobs post by id

### PR DESCRIPTION
**Pull request by uloamaka**
Added the feature for, create, read and update for job post. as they are lesser Go devs, also the e2e test can be run as "go test -v .\tests\jobpost_test.go"

### CODE REVIEW

File: internal/models/job_post.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+func (j *JobPost) EditJobPostById(db *gorm.DB, id string, updates map[string]interface{}) (JobPost, error) {
+	var jobPost JobPost
+	if err := db.First(&jobPost, "id = ?", id).Error; err != nil {
+		return JobPost{}, err
+	}
+	if err := db.Model(&jobPost).Updates(updates).Error; err != nil {
+		return JobPost{}, err
+	}
```
Comment (NwokoyeChigozie) [inline internal/models/job_post.go]: there are functions in postgres repository for this

File: internal/models/job_post.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+	return jobposts, nil
+}
+
+func (j *JobPost) GetJobPostById(db *gorm.DB, id string) (JobPost, error) {
+	var jobpost JobPost
+
+	if err := db.Where("id = ?", id).First(&jobpost).Error; err != nil {
+		return jobpost, err
+	}
```
Comment (NwokoyeChigozie) [inline internal/models/job_post.go]: use the repository please

File: internal/models/job_post.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+
+func (j *JobPost) AddJobPost(db *gorm.DB, jobPost JobPost) (JobPost, error) {
+	jobPost.ID = utility.GenerateUUID()
+	jobPost.CreatedAt = time.Now()
+
+	result := db.Create(&jobPost)
+	if result.Error != nil {
+		return JobPost{}, result.Error
+	}
```
Comment (NwokoyeChigozie) [inline internal/models/job_post.go]: use the repository

File: internal/models/job_post.go [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+	return jobPost, nil
+}
+
+func (j *JobPost) GetAllJobPosts(db *gorm.DB) ([]JobPost, error) {
+	var jobposts []JobPost
+
+	err := db.Find(&jobposts).Error
+	if err != nil {
+		return nil, err
+	}
```
Comment (NwokoyeChigozie) [inline internal/models/job_post.go]: use the repository

File: internal/models/job_post.go [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+	return jobpost, nil
+}
+
+func (j *JobPost) EditJobPostById(db *gorm.DB, id string, updates map[string]interface{}) (JobPost, error) {
+	var jobPost JobPost
+	if err := db.First(&jobPost, "id = ?", id).Error; err != nil {
+		return JobPost{}, err
+	}
+	if err := db.Model(&jobPost).Updates(updates).Error; err != nil {
+		return JobPost{}, err
```
Comment (NwokoyeChigozie) [inline internal/models/job_post.go]: use the repository, I don't want to see db.anything in any model file

File: pkg/controller/jobpost/jobpost.go [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+		Description: body.Description,
+		Location:    body.Location,
+		Salary:      body.Salary,
+		JobType:     body.JobType,
+		CompanyName: body.CompanyName,
+	}
+
+	post, err := jobpost.CreateJobPost(base.Db.Postgresql, jobPost)
+	if err != nil {
+		rd := utility.BuildErrorResponse(http.StatusInternalServerError, "error", "Failed to create job post", err, nil)
```
Comment (NwokoyeChigozie) [inline pkg/controller/jobpost/jobpost.go]: create a service function for business logic

File: pkg/repository/storage/postgresql/connection.go [hunk 7]
```diff
@@ -19,7 +19,7 @@ import (
 func ConnectToDatabase(logger *utility.Logger, configDatabases config.Database) *gorm.DB {
 	dbsCV := configDatabases
 	utility.LogAndPrint(logger, "connecting to database")
-	connectedDB := connectToDb(dbsCV.DB_HOST, dbsCV.USERNAME, dbsCV.PASSWORD, dbsCV.DB_NAME, dbsCV.DB_PORT, dbsCV.SSLMODE, dbsCV.TIMEZONE, logger)
+	connectedDB := connectToDb(dbsCV.DB_HOST, "postgres", dbsCV.PASSWORD, dbsCV.DB_NAME, dbsCV.DB_PORT, dbsCV.SSLMODE, dbsCV.TIMEZONE, logger)
```
Comment (NwokoyeChigozie) [inline pkg/repository/storage/postgresql/connection.go]: please don't try this again

File: pkg/controller/jobpost/jobpost.go [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+	}
+
+	post, err := jobpost.UpdateJobPostById(base.Db.Postgresql, jobPostID, updates)
+	if err != nil {
+		if err == gorm.ErrRecordNotFound {
+			rd := utility.BuildErrorResponse(http.StatusNotFound, "error", "Job post not found", err, nil)
+			c.JSON(http.StatusNotFound, rd)
+			return
+		}
+		rd := utility.BuildErrorResponse(http.StatusInternalServerError, "error", "Failed to update job post", err, nil)
```
Comment (NwokoyeChigozie) [inline pkg/controller/jobpost/jobpost.go]: use service for business logic
Look at what others have done please

**Review by NwokoyeChigozie — CHANGES_REQUESTED** [review]

File: pkg/controller/jobpost/jobpost.go (hunk 8 above)
Comment (uloamaka) [reply pkg/controller/jobpost/jobpost.go]: Ok, thanks for the review. I will effect changes now.

**Pull request closed without merging**